### PR TITLE
Add integration test for smtp functionality in SPI

### LINKED ISSUE #191: Add integration tests for SMTP functionality in SPI

### PR DESCRIPTION
**Pull request by zhongnansu**
Signed-off-by: Zhongnan Su <szhongna@amazon.com>

### Description
Add integration test for smtp functionality in SPI

### Issues Resolved
#191 

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/notifications-test-and-build-workflow.yml [hunk 1]
```diff
@@ -68,7 +68,7 @@ jobs:
       - name: Build with Gradle
         run: |
           cd notifications
-          ./gradlew build -x :integTest -Dopensearch.version=${{ env.OPENSEARCH_VERSION }}.0 # exclude integTest task to avoid permission denied issue
+          ./gradlew build -x :integTest -x :opensearch-notifications-spi:integTest -Dopensearch.version=${{ env.OPENSEARCH_VERSION }}.0 # exclude integTest task to avoid permission denied issue
```
Comment (joshuali925) [inline .github/workflows/notifications-test-and-build-workflow.yml:71]: just curious, what's the permission issue? does integTest run locally?

**Review by joshuali925 — DISMISSED** [review]
LGTM

File: .github/workflows/notifications-test-and-build-workflow.yml (hunk 1 above)
Comment (zhongnansu) [reply .github/workflows/notifications-test-and-build-workflow.yml:71]: It works locally. But in github action, it failed with `java.net.BindException: Permission denied` error when trying the set up the `testSmtpEmailServer`.

you can refer to this to see error logs. Let me know if you have better idea to resolve this so we can add all IT back in CI. https://github.com/opensearch-project/notifications/runs/2970387300?check_suite_focus=true

Comment (zhongnansu) [reply .github/workflows/notifications-test-and-build-workflow.yml:71]: got u, looks similar to what this [post](https://stackoverflow.com/questions/25544849/java-net-bindexception-permission-denied-when-creating-a-serversocket-on-mac-os/25545052) is saying. Though not sure what it works locally, but let me apply the change.

**Review by joshuali925 — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]

**Pull request merged**